lib/pfconfig/memory_cached.pm

```diff
@@ -0,0 +1,96 @@
+package pfconfig::memory_cached;
+
+=head1 NAME
+
+pfconfig::memory_cached
+
+=cut
+
+=head1 DESCRIPTION
+
+pfconfig::memory_cached
+
+This module is used to compute values and tie them to a pfconfig namespace so they are only created when the pfconfig namespace gets updated
+
+This stores the objects in memory (subcache) and flushes them whenever the pfconfig namespace its tied to expires
+
+=cut
+
+=head1 USAGE
+
+=cut
+
+use strict;
+use warnings;
+
+use pf::log;
+use pfconfig::cached;
+our @ISA = ( 'pfconfig::cached' );
+
+=head2 init
+
+Constructor
+
+=cut
+
+sub init {
+    my ( $self, $namespace ) = @_;
+
+    $self->{"_namespace"} = $namespace;
+    $self->{"_control_file_path"} = pfconfig::util::control_file_path($namespace);
+
+    return $self;
+}
+
+=head2 compute
+
+Compute a key using the subcache
+Should the pfconfig namespace have expired, the subcache will fail, meaning we will rebuild.
+
+=cut
+
+sub compute {
+    my ( $self, $key, $sub ) = @_;
+    my $logger = $self->logger;
+
+    my $result = $self->compute_from_subcache($key, sub {
+        return $sub->();
+    });
+
+    return $result;
+}
```

Comment:
Just thought it would be cleaner this way if we add additional logic to the computing.